webapp/src/reducers.ts

```diff
@@ -395,28 +398,60 @@ const voiceUsersStatuses = (state: usersStatusesState = {}, action: usersStatuse
     }
 };
 
+type callRecordingStateAction = {
+    type: string,
+    data: {
+        callID: string,
+        recState: CallRecordingState,
+    },
+}
+
+const callsRecordings = (state: {[callID: string]: CallRecordingState} = {}, action: callRecordingStateAction) => {
+    switch (action.type) {
+    case VOICE_CHANNEL_UNINIT:
+        return {};
+    case VOICE_CHANNEL_CALL_RECORDING_STATE:
+        return {
+            ...state,
+            [action.data.callID]: action.data.recState,
+        };
+    default:
+        return state;
+    }
+};
+
 interface callState {
     channelID: string,
-    startAt: number,
-    ownerID: string,
+    startAt?: number,
+    ownerID?: string,
```

Comment:
Just wondering -- why were these changed to optional?

Reply:
Because we are using the same type but the server can send a partial state (e.g. only the hostID). See `handleCallHostChanged`.